### PR TITLE
new package: py-dateparser

### PR DESCRIPTION
**Pull request by healther**
depends on #13573 #13563 #13569 #13565

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-dateparser/package.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+class PyDateparser(PythonPackage):
+    """dateparser -- python parser for human readable dates"""
+
+    homepage = "https://github.com/scrapinghub/dateparser"
+    url      = "https://pypi.io/packages/source/d/dateparser/dateparser-0.7.2.tar.gz"
+
+    version('0.7.2', sha256='e1eac8ef28de69a554d5fcdb60b172d526d61924b1a40afbbb08df459a36006b')
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-dateparser/package.py]: Add:
```python
depends_on('python@2.7:2.8,3.5:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-dateparser/package.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    depends_on('py-setuptools', type='build')
+    depends_on('py-convertdate@2.1.3:', type=('build', 'run'))
+    depends_on('py-jdatetime@3.1.0:', type=('build', 'run'))
+    depends_on('py-python-dateutil@2.7.5:', type=('build', 'run'))
+    depends_on('py-pytz@2018.9:', type=('build', 'run'))
+    depends_on('py-regex@2019.01.24:', type=('build', 'run'))
+    depends_on('py-tzlocal@1.5.1:', type=('build', 'run'))
+    depends_on('py-umalqurra@0.2:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-dateparser/package.py]: Many of these dependencies are optional. I would either remove them, or add a `+calendars` variant and make them optional. I also don't see any of the version restrictions in the `setup.py`. Where do these come from?

Comment (healther) [reply var/spack/repos/builtin/packages/py-dateparser/package.py]: https://github.com/scrapinghub/dateparser/blob/a08cecce9d57ec88a07687b88fc403bfa5308250/requirements.txt

I just took them from the requirements.txt, iirc this is the one that pip decides on which dependencies to pull in?

File: var/spack/repos/builtin/packages/py-dateparser/package.py (hunk 1 above)
Comment (healther) [reply var/spack/repos/builtin/packages/py-dateparser/package.py]: done

**Comment by healther** [comment]
also added test dependencies and `py-ruamel-yaml` calendar dependency that is only declared in `setup.py` but not in `requirements.txt`. Let's just call my mental state about the compatibility of the python dependency declaration and the zen thingy "there should be exactly one way of doing things" confused...

**Comment by adamjstewart** [comment]
> I just took them from the requirements.txt, iirc this is the one that pip decides on which dependencies to pull in?

Nope, see https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#python-dependencies. `pip` uses whatever is defined in `install_requires` in `setup.py`. In recent versions of setuptools, if the `setup.py` looks oddly empty, it's also good to check in `setup.cfg`.

File: var/spack/repos/builtin/packages/py-dateparser/package.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    depends_on('python@2.7:2.8,3.5:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('py-python-dateutil@2.7.5:', type=('build', 'run'))
+    depends_on('py-pytz@2018.9:', type=('build', 'run'))
+    depends_on('py-regex@2019.01.24:', type=('build', 'run'))
+    depends_on('py-tzlocal@1.5.1:', type=('build', 'run'))
+    depends_on('py-umalqurra@0.2:', type=('build', 'run'), when='+calendars')
+    depends_on('py-ruamel-yaml', type=('build', 'run'), when='+calendars')
+    depends_on('py-convertdate@2.1.3:', type=('build', 'run'), when='+calendars')
+    depends_on('py-jdatetime@3.1.0:', type=('build', 'run'), when='+calendars')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-dateparser/package.py]: I would drop all of the version requirements as they aren't present in the `setup.py`.

Comment (healther) [reply var/spack/repos/builtin/packages/py-dateparser/package.py]: done, I think that we will most likely miss some constraints that are actually required, but not enforced. As the typical use case will be "give me the newest" that likely doesn't matter. However should we put that philosophy in the PythonPackage stub?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-dateparser/package.py]: My philosophy is that if older versions of the dependencies don't work, Spack isn't broken, the `setup.py` is. If I found my version of xyz was too old, I would open an issue with the developers and fix the Spack package later.

Comment (healther) [reply var/spack/repos/builtin/packages/py-dateparser/package.py]: Should I add a couple of comment lines to the PythonPackage stub? Or do you think that is too bloated?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-dateparser/package.py]: Nah, I think the PythonPackage build docs should suffice. I wonder if we should add a link to those in each build system stub...

Comment (healther) [reply var/spack/repos/builtin/packages/py-dateparser/package.py]: That would be helpful, I think I "learned" spack before the doc was in place and didn't think about going back to the documentation

**Pull request merged**